### ISSUE TITLE
CFGF_IGNORE_UNKNOWN as a section-flag

### ISSUE THREAD

**Issue by doublex** [libconfuse/libconfuse#130]
This example, `headers` should support `CFGF_IGNORE_UNKNOWN`:
````
http {
  host = www.example.com
  port = 80
  headers = {
    someheader = 123
    somemore = 456
  }
}
````

**Comment by troglobit** [comment]
The `CFGF_IGNORE_UNKNOWN` flag is supported as an argument to `cfg_init()`, and work as you describe since [v3.1](https://github.com/martinh/libconfuse/releases/tag/v3.1).  We actually have a dedicated unit test for this in [tests/ignore_parm.c](https://github.com/martinh/libconfuse/blob/master/tests/ignore_parm.c).  

Classification: feature request

**Comment by doublex** [comment]
@troglobit 
Thanks for addding `CFGF_KEYSTRVAL`!

Could you add a patch to silence the warning?
````
*** src/confuse.old     2021-04-17 13:51:06.272725596 +0200
--- src/confuse.c       2021-04-17 13:51:43.416710365 +0200
*************** static cfg_opt_t *cfg_getopt_secidx(cfg_
*** 343,349 ****
        if (!index) {
                opt = cfg_getopt_leaf(sec, name);
  
!               if (!opt && !is_set(CFGF_IGNORE_UNKNOWN, cfg->flags))
                        cfg_error(cfg, _("no such option '%s'"), name);
        }
  
--- 343,349 ----
        if (!index) {
                opt = cfg_getopt_leaf(sec, name);
  
!               if (!opt && !is_set(CFGF_IGNORE_UNKNOWN, cfg->flags) && !is_set(CFGF_KEYSTRVAL, cfg->flags))
                        cfg_error(cfg, _("no such option '%s'"), name);
        } 
````

**Comment by troglobit** [comment]
Please post this as a separate issue, or preferably, a pull request.

**Comment by LonerDan** [comment]
Solved by #161, please close.
